packages/vite/src/index.ts

```diff
@@ -122,18 +133,39 @@ export function imagetools(userOptions: Partial<VitePluginOptions> = {}): Plugin
         error: (msg) => this.error(msg)
       }
 
+      const imageBuffer = await img.clone().toBuffer()
+
       for (const config of imageConfigs) {
-        const { transforms } = generateTransforms(config, transformFactories, srcURL.searchParams, logger)
-        const { image, metadata } = await applyTransforms(transforms, img.clone(), pluginOptions.removeMetadata)
+        const id = await generateImageID(srcURL, config, imageBuffer)
+        let image: Sharp | undefined
+        let metadata: ImageMetadata
+
+        if (cacheOptions.enabled && existsSync(`${cacheOptions.dir}/${id}`)) {
+          const imagePath = `${cacheOptions.dir}/${id}`
+          metadata = (await sharp(imagePath).metadata()) as ImageMetadata
+          metadata.imagePath = imagePath
```

Comment:
I'm not sure how I feel about sticking `imagePath` on `metadata`. Could we do something like this instead?
```suggestion
        const id = await generateImageID(srcURL, config, imageBuffer)
        let image: Sharp | undefined
        let imagePath: string | undefined
        let metadata: ImageMetadata

        if (cacheOptions.enabled && existsSync(`${cacheOptions.dir}/${id}`)) {
          imagePath = `${cacheOptions.dir}/${id}`
          metadata = (await sharp(imagePath).metadata()) as ImageMetadata
```